services/task-svc/internal/patient/patient.go

```diff
@@ -0,0 +1,101 @@
+package patient
+
+import (
+	"context"
+	"github.com/google/uuid"
+	zlog "github.com/rs/zerolog/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"hwgorm"
+	"task-svc/api"
+)
+
+type Base struct {
+	HumanReadableIdentifier string `gorm:"column:human_readable_identifier"`
+}
+
+type Patient struct {
+	Base
+	Id uuid.UUID `gorm:"column:id"`
+}
+
+type ServiceServer struct {
+	api.UnimplementedPatientServiceServer
+}
+
+func NewServiceServer() *ServiceServer {
+	return &ServiceServer{}
+}
+
+func (ServiceServer) CreatePatient(ctx context.Context, req *api.CreatePatientRequest) (*api.CreatePatientResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	patient := Patient{
+		Base: Base{
+			HumanReadableIdentifier: req.HumanReadableIdentifier,
+		},
+	}
+
+	result := db.Create(&patient)
+	if err := result.Error; err != nil {
+		log.Warn().Err(err).Msg("database error")
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	log.Info().
+		Str("patientId", patient.Id.String()).
+		Msg("patient created")
+
+	return &api.CreatePatientResponse{
+		Id: patient.Id.String(),
+	}, nil
+}
+
+func (ServiceServer) GetPatient(ctx context.Context, req *api.GetPatientRequest) (*api.GetPatientResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	id, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	patient := Patient{Id: id}
+	result := db.First(&patient)
+	if err := result.Error; err != nil {
```

Comment:
I don't know how this will behave when the query returns empty. This will result in a bad UX, when we respond with an internal server error. Can we handle this case?

I think in hwgorm there is a helper function to separate user and client errors, but it maybe was lost in hw/rest-api